### ISSUE TITLE
Allow snapm output to be formatted in JSON notation

### ISSUE THREAD

**Issue by bmr-cymru** [snapshotmanager/snapm#37]
It should be possible to request the output of snapshot manager commands in JSON notation. This will be useful for automation and integration.

Both the show and list subcommands for snapshot sets and snapshots should allow this output option.

Related: #3

**Comment by bmr-cymru** [comment]
JSON output for show commands is now implemented:

```
root@localhost:~/src/git/snapm# snapm snapset show --members --json
[
    {
        "SnapsetName": "backup",
        "MountPoints": [
            "/",
            "/var",
            "/home"
        ],
        "NrSnapshots": 3,
        "Timestamp": 1717851708,
        "Time": "2024-06-08 14:01:48",
        "UUID": "f666fc16-c9b4-5008-8591-fb5e4eb24386",
        "Status": "Invalid",
        "Snapshots": [
            {
                "Name": "fedora/root-snapset_backup_1717851708_-",
                "SnapsetName": "backup",
                "Origin": "/dev/fedora/root",
                "Timestamp": 1717851708,
                "Time": "2024-06-08 14:01:48",
                "MountPoint": "/",
                "Provider": "lvm2-cow",
                "UUID": "6912de2b-eee9-5e4f-a3ac-43166422e3c0",
                "Status": "Active",
                "Size": "512.0MiB",
                "Free": "492.6MiB",
                "SizeBytes": 536870912,
                "FreeBytes": 516577191,
                "Autoactivate": true,
                "DevicePath": "/dev/fedora/root-snapset_backup_1717851708_-"
            },
            {
                "Name": "fedora/var-snapset_backup_1717851708_-var",
                "SnapsetName": "backup",
                "Origin": "/dev/fedora/var",
                "Timestamp": 1717851708,
                "Time": "2024-06-08 14:01:48",
                "MountPoint": "/var",
                "Provider": "lvm2-cow",
                "UUID": "d241adee-aa01-592f-bbc5-8ce073e8fc86",
                "Status": "Invalid",
                "Size": "512.0MiB",
                "Free": "0B",
                "SizeBytes": 536870912,
                "FreeBytes": 0,
                "Autoactivate": true,
                "DevicePath": ""
            },
            {
                "Name": "fedora/home-snapset_backup_1717851708_-home",
                "SnapsetName": "backup",
                "Origin": "/dev/fedora/home",
                "Timestamp": 1717851708,
                "Time": "2024-06-08 14:01:48",
                "MountPoint": "/home",
                "Provider": "lvm2-thin",
                "UUID": "62ce37f6-edf2-50e1-a045-04722a35d349",
                "Status": "Inactive",
                "Size": "1.0GiB",
                "Free": "1.9GiB",
                "SizeBytes": 1073741824,
                "FreeBytes": 2079623165,
                "Autoactivate": false,
                "DevicePath": ""
            }
        ]
    },
    {
        "SnapsetName": "upgrade",
        "MountPoints": [
            "/",
            "/var",
            "/home"
        ],
        "NrSnapshots": 3,
        "Timestamp": 1717784308,
        "Time": "2024-06-07 19:18:28",
        "UUID": "f04b21ad-28bc-537f-9a8b-a1f5ae3b45ce",
        "Status": "Active",
        "BootEntries": {
            "SnapshotEntry": "a95b5e0",
            "RevertEntry": "be05879"
        },
        "Snapshots": [
            {
                "Name": "fedora/root-snapset_upgrade_1717784308_-",
                "SnapsetName": "upgrade",
                "Origin": "/dev/fedora/root",
                "Timestamp": 1717784308,
                "Time": "2024-06-07 19:18:28",
                "MountPoint": "/",
                "Provider": "lvm2-cow",
                "UUID": "bf0e5d0c-3887-5b10-b2fa-152ff61f09ff",
                "Status": "Active",
                "Size": "8.8GiB",
                "Free": "8.8GiB",
                "SizeBytes": 9437184000,
                "FreeBytes": 9406985011,
                "Autoactivate": true,
                "DevicePath": "/dev/fedora/root-snapset_upgrade_1717784308_-"
            },
            {
                "Name": "fedora/var-snapset_upgrade_1717784308_-var",
                "SnapsetName": "upgrade",
                "Origin": "/dev/fedora/var",
                "Timestamp": 1717784308,
                "Time": "2024-06-07 19:18:28",
                "MountPoint": "/var",
                "Provider": "lvm2-cow",
                "UUID": "cd98ade3-e65f-52b2-81ac-298643203c13",
                "Status": "Active",
                "Size": "12.0GiB",
                "Free": "6.8GiB",
                "SizeBytes": 12884901888,
                "FreeBytes": 7251622782,
                "Autoactivate": true,
                "DevicePath": "/dev/fedora/var-snapset_upgrade_1717784308_-var"
            },
            {
                "Name": "fedora/home-snapset_upgrade_1717784308_-home",
                "SnapsetName": "upgrade",
                "Origin": "/dev/fedora/home",
                "Timestamp": 1717784308,
                "Time": "2024-06-07 19:18:28",
                "MountPoint": "/home",
                "Provider": "lvm2-thin",
                "UUID": "ad6b8c0d-e3ae-56b9-b900-e1beb14b6654",
                "Status": "Active",
                "Size": "1.0GiB",
                "Free": "1.9GiB",
                "SizeBytes": 1073741824,
                "FreeBytes": 2079623165,
                "Autoactivate": true,
                "DevicePath": "/dev/fedora/home-snapset_upgrade_1717784308_-home"
            }
        ]
    }
]
```

```
root@localhost:~/src/git/snapm# snapm snapshot show --json
[
    {
        "Name": "fedora/root-snapset_backup_1717851708_-",
        "SnapsetName": "backup",
        "Origin": "/dev/fedora/root",
        "Timestamp": 1717851708,
        "Time": "2024-06-08 14:01:48",
        "MountPoint": "/",
        "Provider": "lvm2-cow",
        "UUID": "6912de2b-eee9-5e4f-a3ac-43166422e3c0",
        "Status": "Active",
        "Size": "512.0MiB",
        "Free": "492.6MiB",
        "SizeBytes": 536870912,
        "FreeBytes": 516577191,
        "Autoactivate": true,
        "DevicePath": "/dev/fedora/root-snapset_backup_1717851708_-"
    },
    {
        "Name": "fedora/var-snapset_backup_1717851708_-var",
        "SnapsetName": "backup",
        "Origin": "/dev/fedora/var",
        "Timestamp": 1717851708,
        "Time": "2024-06-08 14:01:48",
        "MountPoint": "/var",
        "Provider": "lvm2-cow",
        "UUID": "d241adee-aa01-592f-bbc5-8ce073e8fc86",
        "Status": "Invalid",
        "Size": "512.0MiB",
        "Free": "0B",
        "SizeBytes": 536870912,
        "FreeBytes": 0,
        "Autoactivate": true,
        "DevicePath": ""
    },
    {
        "Name": "fedora/home-snapset_backup_1717851708_-home",
        "SnapsetName": "backup",
        "Origin": "/dev/fedora/home",
        "Timestamp": 1717851708,
        "Time": "2024-06-08 14:01:48",
        "MountPoint": "/home",
        "Provider": "lvm2-thin",
        "UUID": "62ce37f6-edf2-50e1-a045-04722a35d349",
        "Status": "Inactive",
        "Size": "1.0GiB",
        "Free": "1.9GiB",
        "SizeBytes": 1073741824,
        "FreeBytes": 2079623165,
        "Autoactivate": false,
        "DevicePath": ""
    },
    {
        "Name": "fedora/root-snapset_upgrade_1717784308_-",
        "SnapsetName": "upgrade",
        "Origin": "/dev/fedora/root",
        "Timestamp": 1717784308,
        "Time": "2024-06-07 19:18:28",
        "MountPoint": "/",
        "Provider": "lvm2-cow",
        "UUID": "bf0e5d0c-3887-5b10-b2fa-152ff61f09ff",
        "Status": "Active",
        "Size": "8.8GiB",
        "Free": "8.8GiB",
        "SizeBytes": 9437184000,
        "FreeBytes": 9406985011,
        "Autoactivate": true,
        "DevicePath": "/dev/fedora/root-snapset_upgrade_1717784308_-"
    },
    {
        "Name": "fedora/var-snapset_upgrade_1717784308_-var",
        "SnapsetName": "upgrade",
        "Origin": "/dev/fedora/var",
        "Timestamp": 1717784308,
        "Time": "2024-06-07 19:18:28",
        "MountPoint": "/var",
        "Provider": "lvm2-cow",
        "UUID": "cd98ade3-e65f-52b2-81ac-298643203c13",
        "Status": "Active",
        "Size": "12.0GiB",
        "Free": "6.8GiB",
        "SizeBytes": 12884901888,
        "FreeBytes": 7251622782,
        "Autoactivate": true,
        "DevicePath": "/dev/fedora/var-snapset_upgrade_1717784308_-var"
    },
    {
        "Name": "fedora/home-snapset_upgrade_1717784308_-home",
        "SnapsetName": "upgrade",
        "Origin": "/dev/fedora/home",
        "Timestamp": 1717784308,
        "Time": "2024-06-07 19:18:28",
        "MountPoint": "/home",
        "Provider": "lvm2-thin",
        "UUID": "ad6b8c0d-e3ae-56b9-b900-e1beb14b6654",
        "Status": "Active",
        "Size": "1.0GiB",
        "Free": "1.9GiB",
        "SizeBytes": 1073741824,
        "FreeBytes": 2079623165,
        "Autoactivate": true,
        "DevicePath": "/dev/fedora/home-snapset_upgrade_1717784308_-home"
    }
]
```

**Comment by bmr-cymru** [comment]
Resolved in #38, #47, and #50.
